core/circuit.go

```diff
@@ -0,0 +1,263 @@
+package core
+
+import (
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+)
+
+var gridMeterUsed bool // indicates gridmeter is used already fo a circuit to avoid > 1 usage
+
+type Circuit struct {
+	log    *util.Logger
+	uiChan chan<- util.Param
+
+	Name       string     `mapstructure:"name"`       // meaningful name, used as reference in lp
+	MaxCurrent float64    `mapstructure:"maxCurrent"` // the max allowed current of this circuit
+	MeterRef   string     `mapstructure:"meter"`      // Charge meter reference
+	Circuits   []*Circuit `mapstructure:"circuits"`   // sub circuits as config reference
+
+	parentCircuit *Circuit          // parent circuit reference
+	phaseMeter    api.PhaseCurrents // meter to determine phase current
+	vMeter        *VMeter           // virtual meter if no real meter is used
```

Comment:
Gibts eine Chance, das VMeter als extra Property loszuwerden? Für den Loadpoint ist das ja auch nur ein Zähler- wäre schön wenn das auch analog phaseMeter abgefragt werden könnte und es keine Fallunterscheidung bräuchte.

Reply:
Beim Zuweisen des KP zum Circuit muss der Circuit wisssen, ob er einen vMeter und daher den LP als consumer hinzufügen muss. Ich kann versuchen, das in der Config zu verstecken. Siehe Kommentar unten (configure.go)

Reply:
config wär mir nicht wichtig. schön wäre wenn das vmeter ans phasemeter assigned werden könnte. Der Circuit braucht ja nur einen Zähler.